core/src/main/scala/org/ensime/core/Indexer.scala

```diff
@@ -24,10 +24,21 @@ class Indexer(
     LineSourcePositionHelper.fromFqnSymbol(hit)(vfs)
   )
 
-  def oldSearchTypes(query: String, max: Int) =
-    index.searchClasses(query, max).filterNot {
-      name => name.fqn.endsWith("$") || name.fqn.endsWith("$class")
-    }.map(typeResult)
+  def oldSearchTypes(query: String, max: Int): List[TypeSearchResult] = {
+    // Remove $/$class from the end.
+    def strip(fqn: String): String = fqn.replaceAll("\\$(class)*$", "")
+
+    import org.ensime.util.list._
+
+    index.searchClasses(query, max)
```

Comment:
This method is implementing a legacy feature that has been in the wire protocol for years. It might be possible to change the type signature to `List[ClassDef]` without impact, but it could also impact the wire format of the object being serialised... it's the difference between the client expecting a `case class` or a `sealed trait` (which encodes which `case class` is used). You could try changing this signature and then using `: List[TheSuperType]` in the actor response and then finally removing the type assertion, and see if it breaks anything in the clients.